packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js

```diff
@@ -197,6 +150,21 @@ export const useGiftCards = props => {
         setIsCartUpdating
     ]);
 
+    const shouldDisplayCardBalance = useMemo(
+        () =>
+            mostRecentAction === actions.CHECK_BALANCE &&
+            Boolean(balanceResult.data),
+        [mostRecentAction, balanceResult]
+    );
+
+    // We should only display the last card error if the most recent action was apply or check and they had an error
+    const shouldDisplayCardError = useMemo(
```

Comment:
Same here; why the memos?

Reply:
https://github.com/magento/pwa-studio/pull/2462#discussion_r437593106